packages/bpk-component-graphic-promotion/src/BpkGraphicPromo.tsx

```diff
@@ -40,64 +41,66 @@ export const TEXT_ALIGN = {
   start: 'start',
   center: 'center',
   end: 'end',
-};
+} as const;
 
 export type Props = {
-  className: ?string,
-  contentId: ?string,
-  tagline: ?string,
-  headline: string,
-  subheading: ?string,
-  sponsor: ?{
-    label: string,
-    logo: string,
-    altText: string,
-  },
-  buttonText: string,
-  onClick: () => void,
-  invertVertically: boolean,
-  textAlign: TEXT_ALIGN,
-  style: ?{},
+  className?: string | null;
+  contentId?: string | null;
+  tagline?: string | null;
+  headline: string;
+  subheading?: string | null;
+  sponsor?: {
+    label: string;
+    logo: string;
+    altText: string;
+  } | null;
+  buttonText: string;
+  onClick: () => void;
+  invertVertically?: boolean;
+  textAlign: typeof TEXT_ALIGN[keyof typeof TEXT_ALIGN];
+  style?: {};
 };
 
-const constructAriaLabel = ({
-  buttonText,
-  headline,
-  sponsor,
-  subheading,
-  tagline,
-}: Props) => {
-  const text = [];
-  const addText = (value) => value && text.push(value);
+const constructAriaLabel = (
+  buttonText: Props['buttonText'],
+  headline: Props['headline'],
+  sponsor: Props['sponsor'],
+  subheading: Props['subheading'],
+  tagline: Props['tagline'],
+) => {
+  const text: string[] = [];
+  const addText = (value: string) => value && text.push(value);
 
   if (sponsor) {
     addText(sponsor.label);
     addText(sponsor.altText);
-  } else {
+  } else if (tagline) {
     addText(tagline);
   }
+
   addText(headline);
-  addText(subheading);
+
+  if (subheading) {
```

Comment:
Let's keep it in `addText()`, instead of putting the if everywhere.

Reply:
Much nicer to just fix the type of addText since it already handles it, good spot.

https://github.com/Skyscanner/backpack/pull/2535/commits/378cfb846abfbbbabb2828cf46472edee6a310f1